utils-kotlin/src/main/kotlin/org/sonarsource/kotlin/externalreport/common/Translator.kt

```diff
@@ -0,0 +1,289 @@
+/*
+ * SonarSource Kotlin
+ * Copyright (C) 2018-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.externalreport.common
+
+import java.io.StringWriter
+
+object Translator {
+
+    fun escapeHtml4(input: CharSequence) =
+        StringWriter(input.length * 2).apply {
+            (input.indices).forEach { i ->
+                TO_ESCAPE[input[i]]?.let {
+                    write(it)
+                } ?: write(input[i].toString())
+            }
+        }.toString()
```

Comment:
This is a lot more readable than the library implementation, nice!
I have a couple of different suggestions on how we could simplify it even one more step. Since this is not in production and we are only parsing relatively small amounts of data, it doesn't need to be ultra efficient IMO. Considering that, here's my first suggestion:
```suggestion
        input.map { TO_ESCAPE[it] ?: it }.joinToString("")
```
Although I think it's not necessary, if we want to keep the `StringWriter`, we could still simplify it a little:
```suggestion
        StringWriter(input.length * 2).apply {
            input.forEach { char ->
                write(TO_ESCAPE[char] ?: char.toString())
            }
        }.toString()
```
```